tools/ob_admin/backup_tool/ob_admin_backup_validation_ctx.cpp

```diff
@@ -0,0 +1,754 @@
+/**
+ * Copyright (c) 2024 OceanBase
+ * OceanBase CE is licensed under Mulan PubL v2.
+ * You can use this software according to the terms and conditions of the
+ * Mulan PubL v2. You may obtain a copy of Mulan PubL v2 at:
+ *          http://license.coscl.org.cn/MulanPubL-2.0
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY
+ * KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE. See the
+ * Mulan PubL v2 for more details.
+ */
+#include "ob_admin_backup_validation_ctx.h"
+#define CLEAR_LINE "\033[1K\033[0G"
+namespace oceanbase
+{
+namespace tools
+{
+ObAdminTabletAttr::ObAdminTabletAttr()
+    : ls_id_(), data_type_(), sstable_meta_index_(nullptr), tablet_meta_index_(nullptr),
+      macro_block_id_mappings_meta_index_(nullptr), id_mappings_meta_(nullptr)
+{
+}
+ObAdminTabletAttr::~ObAdminTabletAttr()
+{
+  if (OB_NOT_NULL(sstable_meta_index_)) {
+    sstable_meta_index_->~ObBackupMetaIndex();
+    sstable_meta_index_ = nullptr;
+  }
+  if (OB_NOT_NULL(tablet_meta_index_)) {
+    tablet_meta_index_->~ObBackupMetaIndex();
+    tablet_meta_index_ = nullptr;
+  }
+  if (OB_NOT_NULL(macro_block_id_mappings_meta_index_)) {
+    macro_block_id_mappings_meta_index_->~ObBackupMetaIndex();
+    macro_block_id_mappings_meta_index_ = nullptr;
+  }
+  if (OB_NOT_NULL(id_mappings_meta_)) {
+    id_mappings_meta_->~ObBackupMacroBlockIDMappingsMeta();
+    id_mappings_meta_ = nullptr;
+  }
+}
+ObAdminLSAttr::ObAdminLSAttr()
+    : ls_type_(INVALID), ls_meta_package_(), single_ls_info_desc_(nullptr)
+{
+  sys_tablet_map_.create(4, ObModIds::BACKUP);
+}
+ObAdminLSAttr::~ObAdminLSAttr()
+{
+  FOREACH(iter, sys_tablet_map_)
+  {
+    iter->second->~ObAdminTabletAttr();
+    iter->second = nullptr;
+  }
+  sys_tablet_map_.destroy();
+  if (OB_NOT_NULL(single_ls_info_desc_)) {
+    single_ls_info_desc_->~ObSingleLSInfoDesc();
+    single_ls_info_desc_ = nullptr;
+  }
+}
+ObAdminBackupSetAttr::ObAdminBackupSetAttr()
+    : backup_set_store_(nullptr), backup_set_info_desc_(nullptr), ls_inner_tablet_done_(false),
+      minor_tablet_pos_(0), major_tablet_pos_(0), lock_(ObLatchIds::BACKUP_LOCK)
+{
+  ls_map_.create(100, ObModIds::BACKUP);
```

Comment:
create map return error code

Reply:
fixed. added init function for ctx struct